R/02-filter.R

```diff
@@ -59,3 +59,39 @@ qc_filter_zerocounts <- function(gimap_dataset){
   return(list(filter = counts_filter, reportdf = zerocount_df))
 
 }
+
+#' Create a filter for pgRNAs which have a low log2 CPM value for the plasmid/Day 0 sample/time point
+#' @description This function flags and reports which and how many pgRNAs have low log2 CPM values for the plasmid/Day 0 sample/time point
+#' @param gimap_dataset The special gimap_dataset from the `setup_data` function which contains the log2 CPM transformed data
+#' @importFrom magrittr %>%
+#' @importFrom dplyr mutate
+#' @return a named list with the filter `plasmid_filter` specifying which pgRNAs have low plasmid log2 CPM and a report df `plasmid_filter_report` for the number and percent of pgRNA which have a low plasmid log2 CPM
+#' @examples \dontrun{
+#'   qc_filter_plasmid(gimap_dataset)
+#'   
+#'   qc_filter_plasmid(gimap_dataset, cutoff=2)
+#' }
+#'
+
+qc_filter_plasmid <- function(gimap_dataset, cutoff = NULL){
+  plasmid_data <- data.frame(gimap_dataset$transformed_data$log2_cpm[, 1]) %>% `colnames<-`(c("log2_cpm"))
```

Comment:
Same comment here about specifying column names. Can we make this into an argument?

Reply:
I think that functionality is added in PR #35 also